### DISCUSSION TITLE
Macbook pro waiting at %100 like 5 hours!

### DISCUSSION THREAD

**Discussion by MBPplotter** [Chia-Network/chia-blockchain#5601] [Support]
Hi guys and gals!
I have macbook pro 16" 2019 with 16gb ram - 1tb nvme ssd (built in) - i9 (8 core-16 threads)
And i use chia software.

I have a problem about plotting. My computer waits too much at %31 and %100 (like 4 hours)
with this setup i can just run 3 plots in parallel and it takes around 12 hours.

I checked the files and they are still .tmp
I dont understand why my computer waits too much at %31 and %100.
Possible reasons:
1- Mac OS installed on my temporary space. So i need to get external nvme for temp space.
2- My 2tb wd elements desktop hdd connected to the Overhub | Elektron which dont have power adapter. So i need to buy a new usb hub with adapter.
3- My macbook pro also farming.
After your advices i will continue my investments.
Also here is the logs:

Plot size is: 32
Buffer size is: 3390MiB
Using 128 buckets
Using 2 threads of stripe size 65536

Time for phase 1 = 12779.697 seconds. CPU (145.780%) Thu May 20 09:48:08 2021

Time for phase 2 = 4627.041 seconds. CPU (98.720%) Thu May 20 11:05:15 2021
Wrote: 268

Total compress table time: 1325.664 seconds. CPU (96.940%) Thu May 20 13:37:34 2021
Time for phase 3 = 9138.685 seconds. CPU (98.140%) Thu May 20 13:37:34 2021

Time for phase 4 = 596.988 seconds. CPU (99.060%) Thu May 20 13:47:31 2021
Approximate working space used (without final file): 269.305 GiB
Final File size: 101.334 GiB

And it still waiting at %100. Please help me
<img width="1201" alt="Screenshot 2021-05-20 at 16 31 00" src="https://user-images.githubusercontent.com/84523089/119005314-9ff49400-b98f-11eb-97b9-e76b64961008.png">

**Comment by aa2001** [comment]
1. 269.305 (temp) + 101.334 (plot) = 370,639 * 3(thread) = 1 111,917 GiB - Not free space ???
2. 3 threads at the same time finished creating the plot, and copy the data to the ext hdd, which will take a long time.

370,639 GiB is used in the final stage,You need to make a delay between starting the threads.

The delay time is the total time to create a plot / 3.

**Reply by MBPplotter** [reply]
so u say
12779 + 4627 + 9138 + 596 = 27140
27140/3 = 9046
9046/60 = 150 min delay between parallel plots ?

**Reply by aa2001** [reply]
Yes.
The first thread finishes running, and starts writing to disk.
The second thread is about 60% of the work.
The third one is just starting 30% work.

**Reply by MBPplotter** [reply]
thank you :)  i will try 153 mins tomorrow and let you know the results i hope i wont exceed the ssd space

**Reply by MBPplotter** [reply]
I want to be sure that my settings are ok for 2tb hdd. Can you check please ?
I9-9880H - 8core 16 thread
16gb ram
850 gb empty space on temporary ssd

I can increase the ram and thread numbers but i dont know if it effect or not :(

<img width="1000" alt="Screenshot 2021-05-21 at 10 45 19" src="https://user-images.githubusercontent.com/84523089/119109786-c3681f00-ba21-11eb-8212-066de251d988.png">

**Reply by aa2001** [reply]
Don't change anything. Threads will increase the sorting speed and break the pauses between threads.
If 850 Gb is enough, then you can experiment later.

**Reply by MBPplotter** [reply]
Update: 
Time for phase 1 = 9909.803 seconds. CPU (147.900%) Fri May 21 11:13:46 2021

Phase 1 finished for first plot 50 mins faster :) I hope total time for 3 plots in parallel will decrease.

Current status:

<img width="965" alt="Screenshot 2021-05-21 at 11 22 26" src="https://user-images.githubusercontent.com/84523089/119115258-40e25e00-ba27-11eb-91bb-154b42347516.png">

**Reply by aa2001** [reply]
To do everything accurately, you need to run the threads with your hands :(
We remember the start time of the first thread, and when it reaches 33% of the launches of the second and remember how much time has passed. When the second thread reaches 33%, the third one starts, and we record how much time it took. There will be a difference, since the two threads will load the SSD more heavily.

**Reply by MBPplotter** [reply]
so second plot should start at %33 of plotting stage for first plot. but why %33 ? I read some discussions and ppl said after 1st phase of 1st plot, second plot should start. As i see first phase finished at %31 (if im not wrong). And my process stuck at %31 too much :( So should it start at %31 or %33 ?

**Reply by aa2001** [reply]
When 1 thread reaches 99%, the second will be at 66%, the third at 33%. Then the volume of temporary files will be about the same - 33+66+99 = 200% * 370,639 = 740 Mib. As soon as 1 thread reaches 100%, it will delete all temporary files, which will allow the rest of the thread to take this place.

**Reply by MBPplotter** [reply]
ok i deleted the other plots and i am waiting for %33 to start second one after the experiment i will note the durations and make parallel plotting :) thank you

**Reply by MBPplotter** [reply]
I started second plot manually when first plot reached %33.
here is the results: I did not wait at %100 (thank you)
but we have some timing problem because when first one reached %99 the second one was at %21 :(

**Comment by BrandtH22** [comment]
@MBPplotter , the newest version(s) of chia has fixes in place to resolve this issue. Please download it from here: https://www.chia.net/downloads/

Since this issue has been resolved with the latest version(s) of chia we will be closing this ticket but if we have closed this ticket in error do not hesitate to reach out to us again with any followup questions, comments, or if the issue persists after an update.

```
Please note that upgrading major versions of chia generally requires the config.yaml to be recreated using these steps:
1. Stop chia
2. Rename the config or move it to a different directory
3. In CLI run chia init
4. Copy any custom settings from the old config to the new (plot drive directories, pool info, trusted node info, etc) - be careful here as yaml is very picky about formatting including those leading spaces
5. Start Chia
```

The best place to reach our support team is on Discord (https://discord.gg/chia) or by reopening this ticket.
